iree/hal/metal/metal_pipeline_argument_buffer.h

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IREE_HAL_METAL_METAL_PIPELINE_ARGUMENT_BUFFER_H_
+#define IREE_HAL_METAL_METAL_PIPELINE_ARGUMENT_BUFFER_H_
+
+#include <string>
+
+#include "absl/container/inlined_vector.h"
+#include "absl/types/span.h"
+#include "iree/hal/descriptor_set.h"
+#include "iree/hal/descriptor_set_layout.h"
+#include "iree/hal/executable_layout.h"
+
+// Metal implementaion classes for resource descriptor related interfaces.
+//
+// A descriptor is an opaque handle pointing to a resource that are accessed in
```

Comment:
Large comment here that's duplicated in the markdown file. Want to just link to the doc from here instead?